pkg/controllers/daemon/service/controller.go

```diff
@@ -74,9 +74,12 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 		return ctrl.Result{}, nil
 	}
+	// ignore headless services
+	if service.Spec.ClusterIP == "None" {
+		return ctrl.Result{}, nil
+	}
```

Comment:
do we want to always, unconditionally, ignore headless services? 
if so, maybe we could filter the events out before they even hit the reconciler.
I haven't evaluated how possible it is for this Object type and field, but [here's how we have done some event filters](https://github.com/Azure/azure-container-networking/blob/251959ac63e361898baa5089f29a548a120fc044/cns/kubecontroller/nodenetworkconfig/reconciler.go#L164-L186) or [here's how we have filtered even earlier in the ctrlmanager cache layer](https://github.com/Azure/azure-container-networking/blob/251959ac63e361898baa5089f29a548a120fc044/cns/service/main.go#L1328-L1343).

Reply:
according to the #547 the expected behavior is to ignore headless services. I have opted for the event filters implementation in [pkg/controllers/daemon/service/controller.go](https://github.com/microsoft/retina/pull/763/files#diff-81cc88e0d83f3f5284a95530b0781f2dbb805aaf11c7f5f0f55e57b30978cba3)